js_modules/dagster-ui/packages/ui-components/src/theme/GlobalThemeStyle.tsx

```diff
@@ -0,0 +1,40 @@
+import {createGlobalStyle} from 'styled-components';
+
+import {darkThemeColors} from './darkThemeColors';
+import {lightThemeColors} from './lightThemeColors';
+import {CoreColorStyles} from '../palettes/CoreColorStyles';
+import {DataVizColorStyles} from '../palettes/DataVizColorStyles';
+import {TranslucentColorStyles} from '../palettes/TranslucentColorStyles';
+
+const ThemeRoot = createGlobalStyle`
+  @media (prefers-color-scheme: light) {
+    :root, .themeSystem {
+      ${lightThemeColors}
+    }
+  }
+
+  @media (prefers-color-scheme: dark) {
+    :root, .themeSystem {
+      ${darkThemeColors}
+    }
+  }
+
+  .themeLight {
+    ${lightThemeColors}
+  }
+
+  .themeDark {
+    ${darkThemeColors}
+  }
+`;
```

Comment:
a shame that this is going to be a really giant string. I think with https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries we could reduce that but the browser support is still rather limited :|